rust-src/concordium_base/src/transactions.rs

```diff
@@ -1464,6 +1465,52 @@ pub struct AccountAccessStructure {
     pub threshold: AccountThreshold,
 }
 
+type AccountStructure<'a> = &'a [(
+    CredentialIndex,
+    SignatureThreshold,
+    &'a [(KeyIndex, ed25519_dalek::PublicKey)],
+)];
+
+impl AccountAccessStructure {
+    /// Generate a new [`AccountAccessStructure`] for the thresholds, public
+    /// keys, and key indices specified in the input. If there are duplicate
+    /// indices then later ones override the previous ones.
+    pub fn generate(account_threshold: AccountThreshold, structure: AccountStructure) -> Self {
+        let mut map: BTreeMap<CredentialIndex, CredentialPublicKeys> = BTreeMap::new();
+
+        for credential_structure in structure {
+            let mut inner_map: BTreeMap<KeyIndex, VerifyKey> = BTreeMap::new();
+
+            for key_structure in credential_structure.2 {
+                inner_map.insert(
+                    key_structure.0,
+                    VerifyKey::Ed25519VerifyKey(key_structure.1),
+                );
+            }
+
+            map.insert(credential_structure.0, CredentialPublicKeys {
+                keys:      inner_map,
+                threshold: credential_structure.1,
+            });
+        }
+
+        AccountAccessStructure {
+            keys:      map,
+            threshold: account_threshold,
+        }
+    }
+
+    /// Generate a new [`AccountAccessStructure`] with a single credential and
+    /// public key, at credential and key indices 0.
```

Comment:
```suggestion
    /// public key, at credential and key index 0.
```

Reply:
I feel it should be plural because we refer to both indices `credential index 0` and the `key index 0`.